spec/models/vacation_spec.rb

```diff
@@ -151,6 +151,33 @@
     end
   end
 
+  describe '.finished' do
+    let!(:ended_vacations) { create_list(:vacation, 3, :ended) }
+    let(:ended_vacation) { create(:vacation, :ended) }
+
+    before do
+      create(:vacation, :approved)
+      create(:vacation, :pending)
+      create(:vacation, :cancelled)
+      create(:vacation, :denied)
+      create(:vacation, :ongoing)
+      create(:vacation, :scheduled)
+      create(:vacation, :expired)
+      create(:vacation, :valid)
+    end
+    
+    it 'returns only finished vacations' do
+      expect(described_class.finished).to eq(ended_vacations)
+    end
+
+    it 'returns finished vacations sorted by most recent' do
+      ended_vacation.update(end_date: 2.days.ago)
+      sorted_ended_vacations = ended_vacations.push(ended_vacation)
+
+      expect(described_class.finished).to eq(sorted_ended_vacations)
+    end
+  end
```

Comment:
```suggestion
  describe '.finished' do
    let!(:ended_vacation1) { create(:vacation, :ended, end_date: 1.day.ago) }
    let!(:ended_vacation2) { create(:vacation, :ended, end_date: 2.days.ago) }
    let!(:ended_vacation3) { create(:vacation, :ended, end_date: 3.days.ago) }

    before do
      create(:vacation, :approved)
      create(:vacation, :pending)
      create(:vacation, :cancelled)
      create(:vacation, :denied)
      create(:vacation, :ongoing)
      create(:vacation, :scheduled)
      create(:vacation, :expired)
      create(:vacation, :valid)
    end
    
    it 'returns only finished vacations sorted by most recent' do
      expect(described_class.finished).to eq([ended_vacation1, ended_vacation2, ended_vacation3])
    end
  end
```

Reply:
I thought that would look ugly but between two uglies better pick the lesser ugly :^). I have also removed that redundant test.